File: DataBindingHandlerJava/app/src/main/java/com/websarva/wings/android/databindinghandlerjava/ui/MainViewModel.java
package com.websarva.wings.android.databindinghandlerjava.ui;

import androidx.lifecycle.ViewModel;

import com.websarva.wings.android.databindinghandlerjava.data.model.Rectangle;

public class MainViewModel extends ViewModel {
	Rectangle _rectangle;

	public MainViewModel() {
		generateNewRectangle();
	}

	public void generateNewRectangle() {
		int height = (int) (Math.random() * 10) + 1;
		int width = (int) (Math.random() * 10) + 1;
		_rectangle = new Rectangle(height, width);
	}

	public Rectangle getRectangle() {
		return _rectangle;
	}
}

File: DataBindingHandlerKotlin/app/src/main/java/com/websarva/wings/android/databindinghandlerkotlin/ui/MainViewModel.kt
package com.websarva.wings.android.databindinghandlerkotlin.ui

import androidx.lifecycle.ViewModel
import com.websarva.wings.android.databindinghandlerkotlin.data.model.Rectangle

class MainViewModel : ViewModel() {
	var rectangle: Rectangle

	init {
		rectangle = Rectangle(0, 0)
		generateNewRectangle()
	}

	fun generateNewRectangle() {
		val height = (Math.random() * 10).toInt() + 1
		val width = (Math.random() * 10).toInt() + 1
		rectangle = Rectangle(height, width)
	}
}
